### PR TITLE
Cosmitar refactory components

### PR DESCRIPTION
**Pull request by Cosmitar**
#### What does this PR do?
- Adds React components structure. 
- Refactory of App.js.
- Includes views state: Home - Search Results - Player.
- Reduce the number of search results to 20.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,42 +1,69 @@
...
-            <span>{this.props.searchText}</span>
-            <span>{this.props.resultList.length}</span>
-            <button onClick={this._handleStartPlaying.bind(this)} style={triggerDisplayStyle}>
-                Start TV Channel
-            </button>
-            {player}
-        </div>;
+        return (
+            <div className='app-inner-wrapper'>
+                { this.props.view == VIEWS.HOME ?       this._getHomeLayout()     : null }
```
Comment (loverajoel) [inline src/App.js]: remove this strange spaces, this is not php :dancer:

File: src/App.js [hunk 2]
```diff
@@ -1,42 +1,69 @@
...
         );
     }
 
-    _handleStartPlaying() {
-        this.props.openPlayer();
+    _getResultsLayout() {
+        return (
+            <div className='results-wrapper'>
+                <Header>
+                    <Search></Search>
```
Comment (loverajoel) [inline src/App.js]: **NonBloker**
you can do `<Search/>` more simple

File: src/components/footer/Footer.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+'use strict';
```
Comment (loverajoel) [inline src/components/footer/Footer.js]: this component should be a [**stateless functional components**](https://facebook.github.io/react/blog/2015/10/07/react-v0.14.html#stateless-functional-components)

File: src/components/header/Header.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+'use strict';
```
Comment (loverajoel) [inline src/components/header/Header.js]: same here

File: src/components/search/Search.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
+'use strict';
+import React, {Component} from 'react';
+import {connect} from 'react-redux';
+import * as searchActions from './../../actions/search';
+
+class Search extends Component {
+    render() {
+        return (
+            <div className={'search-wrapper' + (this.props.isSearching ? 'search-mask' : '') }>
+                <input type='text' ref='searchBox'
```
Comment (loverajoel) [inline src/components/search/Search.js]:
```
<input
    type='text'
    ref='searchBox'
    onKeyPress={this._handleKeyPress.bind(this)}
/>
```

**Comment by loverajoel** [comment]
works as expected!  after a couple of minors changes :shipit:

File: src/App.js (hunk 1 above)
Comment (Cosmitar) [inline src/App.js]: Readability doesn't count? Your linter configuration says it's valid code ;)

Comment (loverajoel) [inline src/App.js]: :-1:

File: src/App.js (hunk 2 above)
Comment (Cosmitar) [inline src/App.js]: ok, I undestand

**Pull request merged**